daisy_workflows/export_metadata/export-metadata.ps1

```diff
@@ -0,0 +1,86 @@
+function Get-MetadataValue {
+    param (
+        [parameter(Mandatory=$true)]
+        [string]$key,
+        [parameter(Mandatory=$false)]
+        [string]$default
+    )
+
+    # Returns the provided metadata value for a given key.
+    $url = "http://metadata.google.internal/computeMetadata/v1/instance/attributes/$key"
+    $max_attemps = 30
+    for ($i=0; $i -le $max_attemps; $i++) {
+        try {
+            $client = New-Object Net.WebClient
+            $client.Headers.Add('Metadata-Flavor', 'Google')
+            $value = ($client.DownloadString($url)).Trim()
+            Write-Host "Retrieved metadata for key $key with value $value."
+            return $value
+        }
+        catch [System.Net.WebException] {
+            if ($default) {
+                Write-Host "Failed to retrieve metadata for $key, returning default $default."
+                return $default
+            }
+            # Sleep after each failure with no default value to give the network adapters time to become functional.
+            Start-Sleep -s 1
+        }
+    }
+    Write-Host "Failed $max_attemps times to retrieve value from metadata for $key, returning null."
+    return $null
+}
+
+function Export-ImageMetadata {
+    $image_version = Get-Date -Format "yyyyMMdd"
+    $build_date = Get-Date -Format "o"
+    $image_metadata = @{'id' = $image_id;
+                        'name' = $image_name;
+                        'family' = $image_family;
+                        'version' = $image_version;
+                        'build_date' = $build_date;
+                        'packages' = @()}
+
+    # Get Googet packages.
+    $out = & 'C:\ProgramData\GooGet\googet.exe' -root 'C:\ProgramData\GooGet' 'installed'
+    $out = $out[1..$out.length]
+    [array]::sort($out)
+
+    foreach ($package_line in $out) {
+        try {
+            $name = $package_line.Trim().Split(' ')[0]
+            # Get Package Info for each package
+            $info = & 'C:\ProgramData\GooGet\googet.exe' -root 'C:\ProgramData\GooGet' 'installed' '-info' $name
+            $version = $info[4].Split(":")[1].Trim()
```

Comment:
Just indexing the output by lines can easily break if the output is altered, instead you should proabbly scan the output line by line looking for the correct line based on prefix.

Reply:
Great catch. fix.